### PR TITLE
DEV-1968: Avatar Menu

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a

#### What's this PR do?
Creates Avatar Menu component and add it to the top header/nav. Avatar Menu shows the user's initials and when clicked pop up several additional settings & options for the user (Example: sign out, FAQ link)

#### Why are we doing this? How does it help us?
Makes it easier for the user to understand what account they are logged in, and simplifies user interaction with settings.

#### How should this be manually tested? Please include detailed step-by-step instructions.
Design
- Access storybook `Components > AvatarMenu` and check that the design matches all use-cases and menu options.
Usability
- Access the dashboard with a logged in user
- Make sure avatar menu shows with the user's initials (probably will use the email's first letter)
- Click FAQ link and open in new tab
- Click logout and make sure user logs out

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
yes

#### Have automated unit tests been added? If not, why?
yes

#### How should this change be communicated to end users?
User menu now added with some functionality and will be added more in the future.

#### Are there any smells or added technical debt to note?
No

#### Has this been documented? If so, where?
no

#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-1968

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
No

### CODE REVIEW

File: spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    tree({ user: emailUser });
+
+    userEvent.click(screen.getByRole('button', { name: settingsMenu }));
+    expect(screen.getByRole('menuitem', { name: 'FAQ' })).toBeEnabled();
+    expect(screen.getByRole('menuitem', { name: 'Sign out' })).toBeEnabled();
+  });
+
+  it('should have correct FAQ link', () => {
+    window.open = jest.fn()
```
Comment (nrh-cklimas) [inline spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx:75]: We should restore this after the test finishes. Something like...

```ts
const oldOpen = jest.spyOn(window, 'open').mockImplementation(() => {});
// ... test ...
oldOpen.mockRestore();
```

File: spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    userEvent.click(screen.getByRole('button', { name: settingsMenu }));
+    userEvent.click(screen.getByRole('menuitem', { name: 'Sign out' }));
+    expect(onLogout).toBeCalledTimes(1)
+  });
+
+  it('is accessible', async () => {
+    const { container } = tree({ user: { email: 'a@a.com' } });
+    expect(await axe(container)).toHaveNoViolations();
+
+    userEvent.click(screen.getByRole('button', { name: settingsMenu }));
```
Comment (nrh-cklimas) [inline spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx:102]: 👍  checking AX both closed and open.

File: spa/src/components/common/AvatarMenu/AvatarMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    setAnchorEl(null);
+  };
+
+  const handleFAQ = () => {
+    window.open(urlFAQ, '_blank', 'noopener, noreferrer');
+  };
+
+  return (
+    <>
+      <Container open={open ? 'open' : ''} className={className!} onClick={handleClick} aria-label='settings menu'>
```
Comment (nrh-cklimas) [inline spa/src/components/common/AvatarMenu/AvatarMenu.tsx:42]: How about just `Settings` for the ARIA label to match what you put on ModalHeader?

File: spa/src/components/common/AvatarMenu/AvatarMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import onLogout from 'components/authentication/logout';
+
+import {
+  Container, ModalHeader, Popover, Avatar, ListWrapper,
+  MenuItem, ListItemIcon, Typography, LogoutIconWrapper
+} from './AvatarMenu.styled';
+
+export type AvatarMenuProps = InferProps<typeof AvatarMenuPropTypes>
+
+export const urlFAQ = 'https://news-revenue-hub.atlassian.net/servicedesk/customer/portal/11/article/2195423496'
```
Comment (nrh-cklimas) [inline spa/src/components/common/AvatarMenu/AvatarMenu.tsx:16]: We have this and the help URL already in DashboardSidebarFooter. How about we centralize them under `constants` so they don't fall out of sync?

File: spa/src/components/common/AvatarMenu/AvatarMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+export const capitalizeInitial = (text?: string) => text ? text[0].toUpperCase() : ''
+
+const AvatarMenu = ({ user, className }: AvatarMenuProps) => {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const id = open ? 'grab link popover' : undefined;
```
Comment (nrh-cklimas) [inline spa/src/components/common/AvatarMenu/AvatarMenu.tsx:22]: Let's make this something like `avatar-menu-popover`.

File: spa/src/components/common/AvatarMenu/AvatarMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    email: PropTypes.string.isRequired
+  }),
+  className: PropTypes.string
+};
+
+AvatarMenu.propTypes = AvatarMenuPropTypes;
+
+AvatarMenu.defaultProps = {
+  className: '',
+  user: undefined,
```
Comment (nrh-cklimas) [inline spa/src/components/common/AvatarMenu/AvatarMenu.tsx:93]: I don't think setting a default prop to undefined has any effect?

File: spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+import { render, screen, within } from 'test-utils';
+import userEvent from '@testing-library/user-event';
+
+import onLogout from 'components/authentication/logout';
+import AvatarMenu, { AvatarMenuProps, urlFAQ } from './AvatarMenu';
+
+jest.mock('components/authentication/logout', () => ({
+  __esModule: true,
+  default: jest.fn()
+}));
```
Comment (nrh-cklimas) [inline spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx:11]:
```suggestion
jest.mock('components/authentication/logout');
```

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]
Overall looks good, just had some small suggestions.

Could you run prettier on this PR? I noticed some stuff it should clean up -- missing semicolons, some inconsistencies about single/double quotes.

File: spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx (hunk 1 above)
Comment (GuiMend) [reply spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx:75]: 👍 
What is the issue with the current implementation? Would like to learn why it's not ideal

File: spa/src/components/common/AvatarMenu/AvatarMenu.tsx (hunk 6 above)
Comment (GuiMend) [reply spa/src/components/common/AvatarMenu/AvatarMenu.tsx:93]: Correct

File: spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx (hunk 1 above)
Comment (nrh-cklimas) [reply spa/src/components/common/AvatarMenu/AvatarMenu.test.tsx]: It's kind of a theoretical problem here, but changing a global in a Jest test can cause it to pollute other tests if they rely on it without mocking it (which they shouldn't... but nobody's perfect). And this usually shows up as weird test failures dependent on which tests are running, order in which they run, etc.

**Review by nrh-cklimas — APPROVED** [review]

**Review by x110dc — APPROVED** [review]

**Pull request merged**